L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryV9Imp3.cc

```diff
@@ -0,0 +1,1095 @@
+    links = links_per_module_.at(packed_module);
+  }
+  return links;
+}
+
+unsigned HGCalTriggerGeometryV9Imp3::getModuleSize(const unsigned module_id) const {
+  unsigned nWafers = 1;
+  return nWafers;
+}
+
+unsigned HGCalTriggerGeometryV9Imp3::getNextSector(const unsigned sector) const {
+  unsigned next_sector = 0;
+  if (sector < 2) {
+    next_sector = sector + 1;
+  }
+  return next_sector;
+}
+
+unsigned HGCalTriggerGeometryV9Imp3::getPreviousSector(const unsigned sector) const {
+  unsigned previous_sector = 2;
+  if (sector > 0) {
+    previous_sector = sector - 1;
+  }
+  return previous_sector;
+}
+
+HGCalTriggerGeometryBase::geom_set HGCalTriggerGeometryV9Imp3::getStage1FpgasFromStage2Fpga(
+    const unsigned stage2_id) const {
+  geom_set stage1_ids;
+  HGCalBackendDetId id(stage2_id);
+
+  geom_set stage1_links = getStage1LinksFromStage2Fpga(stage2_id);
+  for (const auto& stage1_link : stage1_links) {
+    stage1_ids.emplace(getStage1FpgaFromStage1Link(stage1_link));
+  }
+
+  return stage1_ids;
+}
+
+HGCalTriggerGeometryBase::geom_set HGCalTriggerGeometryV9Imp3::getStage2FpgasFromStage1Fpga(
+    const unsigned stage1_id) const {
+  geom_set stage2_ids;
+  HGCalBackendDetId id(stage1_id);
+
+  geom_set stage1_links = getStage1LinksFromStage1Fpga(stage1_id);
+  for (const auto& stage1_link : stage1_links) {
+    stage2_ids.emplace(getStage2FpgaFromStage1Link(stage1_link));
+  }
+
+  return stage2_ids;
+}
+
+HGCalTriggerGeometryBase::geom_set HGCalTriggerGeometryV9Imp3::getStage1LinksFromStage2Fpga(
+    const unsigned stage2_id) const {
+  geom_set stage1link_ids;
+  HGCalBackendDetId id(stage2_id);
+
+  auto stage2_itrs = stage2_to_stage1links_.equal_range(id.label());
+  for (auto stage2_itr = stage2_itrs.first; stage2_itr != stage2_itrs.second; stage2_itr++) {
+    if (stage2_itr->second == true) {  //link and stage2 FPGA are the same sector
+      stage1link_ids.emplace(
+          HGCalBackendDetId(id.zside(), HGCalBackendDetId::BackendType::Stage1Link, id.sector(), stage2_itr->second));
+    } else {  //link is from the next sector (anti-clockwise)
+      stage1link_ids.emplace(HGCalBackendDetId(
+          id.zside(), HGCalBackendDetId::BackendType::Stage1Link, getNextSector(id.sector()), stage2_itr->second));
+    }
+  }
+
+  return stage1link_ids;
+}
+
+unsigned HGCalTriggerGeometryV9Imp3::getStage1FpgaFromStage1Link(const unsigned link_id) const {
+  HGCalBackendDetId id(link_id);
+  unsigned stage1_label = stage1link_to_stage1_.at(id.label());
+
+  return HGCalBackendDetId(id.zside(), HGCalBackendDetId::BackendType::Stage1FPGA, id.sector(), stage1_label);
+}
+
+unsigned HGCalTriggerGeometryV9Imp3::getStage2FpgaFromStage1Link(const unsigned link_id) const {
+  HGCalBackendDetId id(link_id);
+  bool same_sector = stage1link_to_stage2_.at(id.label());
+  unsigned sector = id.sector();
+
+  if (same_sector == false) {
```

Comment:
Would be shorter to write simply `if (!same_sector)`

Reply:
done